pkg/cache/cache.go

```diff
@@ -287,20 +287,17 @@ func (options Options) inheritFrom(inherited Options) (*Options, error) {
 	return &combined, nil
 }
 
-func combineScheme(schemes ...*runtime.Scheme) *runtime.Scheme {
-	var out *runtime.Scheme
-	for _, sch := range schemes {
-		if sch == nil {
-			continue
-		}
-		for gvk, t := range sch.AllKnownTypes() {
-			if out == nil {
-				out = runtime.NewScheme()
-			}
-			out.AddKnownTypeWithName(gvk, reflect.New(t).Interface().(runtime.Object))
-		}
+func combineScheme(inherited *runtime.Scheme, new *runtime.Scheme) *runtime.Scheme {
+	if inherited == nil {
+		return new
+	}
+	if new == nil {
+		return inherited
+	}
+	for gvk, t := range new.AllKnownTypes() {
+		inherited.AddKnownTypeWithName(gvk, reflect.New(t).Interface().(runtime.Object))
 	}
-	return out
+	return inherited
```

Comment:
Won't this leak `new` scheme types into the inherited scheme? What happens here if (somehow) a conflicting scheme is required by a different cache, and that cache shares the same inherited scheme? Will we change the scheme out from under the first cache?